managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1868,7 +1868,15 @@ public void asyncMarkDelete(final Position position, Map<String, Long> propertie
 
         if (config.isDeletionAtBatchIndexLevelEnabled() && batchDeletedIndexes != null) {
             if (newPosition.ackSet != null) {
-                batchDeletedIndexes.put(newPosition, BitSetRecyclable.create().resetWords(newPosition.ackSet));
+                lock.writeLock().lock();
+                try {
+                    BitSetRecyclable givenBitSet = BitSetRecyclable.create().resetWords(newPosition.ackSet);
+                    batchDeletedIndexes.compute(newPosition,
+                            (k, v) -> v == null || givenBitSet.nextSetBit(0) > v.nextSetBit(0)
+                                    ? givenBitSet : v);
```

Comment:
I am curious why not use the result of `givenBitSet & v` instead of choosing between them?

Reply:
For example, the received batch index binary digits are 111000, 11000, and 101000 and 100001 cannot appear：
<img width="1481" alt="image" src="https://user-images.githubusercontent.com/19296967/196354078-c6727497-0e88-4f13-8370-f5d1c9b33186.png">

Reply:
The cumulative ack of batch index, 0 means confirmation, if we want to confirm the sub-message with batch index of 3, the batch size is 10, the corresponding binary is 1111111000, and the binary bits of the sub-message index before the confirmed batch index will also be filled.